src/util/query.js

```diff
@@ -37,14 +38,19 @@ module.exports.build = (allowedFields, {
     result.query = buildQuery(filterBy, allowedFields);
   }
   if (scoreBy.length !== 0) {
-    result.query = {
-      function_score: {
-        query: get(result, 'query', { match_all: {} }),
-        functions: scoreBy.map(e => actionMap.score[e[0]](...e.slice(1))),
-        score_mode: 'sum',
-        boost_mode: 'replace'
-      }
-    };
+    set(result, 'query.bool.should', []);
+    result.query.bool.should.push(
+      { function_score: { script_score: { script: { source: '0' } } } },
+      ...scoreBy.map(s => (s[1].includes('.')
+        ? ({
+          nested: {
+            path: s[1].substring(0, s[1].lastIndexOf('.')),
+            query: { function_score: actionMap.score[s[0]](s.slice(1), { allowedFields }) },
+            score_mode: 'max'
+          }
+        })
+        : ({ function_score: actionMap.score[s[0]](s.slice(1), { allowedFields }) })))
```

Comment:
Not sure how I feel about the redundancy. Can we just use map and then `.map(([path, query]) => (path.includes('.') ? ... : ...)`?

Reply:
I went a slightly different way, because the mapping to `path` strips out the last `.`, and therefore the information around if this is a nested path or not.
Which meant I would need to pass through that information as `s` anyway, it didn't quite looks as clean in practice